pkg/ext/stores/useractivity/store.go

```diff
@@ -0,0 +1,203 @@
+			return nil, apierrors.NewInternalError(fmt.Errorf("failed to get token %s: %v", objUserActivity.Spec.TokenID, err))
+		}
+	}
+	// set when last activity happened
+	lastActivity := metav1.Time{
+		Time: time.Now().UTC(),
+	}
+	// retrieve setting for auth-user-session-idle-ttl-minutes
+	idleTimeout := settings.AuthUserSessionIdleTTLMinutes.GetInt()
+
+	// check if it's a dry-run
+	dryRun := options != nil && len(options.DryRun) > 0 && options.DryRun[0] == metav1.DryRunAll
+
+	return uas.create(ctx, objUserActivity, token, token.UserID, lastActivity, idleTimeout, dryRun)
+}
+
+// create sets the LastActivity and CurrentTimeout fields on the UserActivity object
+// provided by the user within the request.
+func (uas *Store) create(_ context.Context,
+	userActivity *ext.UserActivity,
+	token *v3Legacy.Token,
+	user string,
+	lastActivity metav1.Time,
+	authUserSessionIdleTTLMinutes int,
+	dryRun bool) (*ext.UserActivity, error) {
+
+	expectedName, err := setUserActivityName(user, token.Name)
+	if err != nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("failed to set useractivity name: %v", err))
+	}
+	// ensure the UserActivity object is crafted as expected.
+	if userActivity.Name != expectedName {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("useractivity name mismatch: have %s - expected %s", userActivity.Name, expectedName))
+	}
+	// ensure the token specified in the UserActivity is the same
+	// we are using to do the request.
+	if token.Name != userActivity.Spec.TokenID {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("token name mismatch: have %s - expected %s", token.Name, userActivity.Spec.TokenID))
+	}
+
+	// once validated the request, we can define the lastActivity time.
+	newIdleTimeout := metav1.Time{
+		Time: lastActivity.Add(time.Minute * time.Duration(authUserSessionIdleTTLMinutes)).UTC(),
+	}
+	userActivity.Status.LastActivity = lastActivity.String()
+	userActivity.Status.CurrentTimeout = newIdleTimeout.String()
+
+	// if it's not a dry-run, commit the changes
+	if !dryRun {
+		token.LastIdleTimeout = newIdleTimeout
+		_, err = uas.tokenController.Update(token)
+		if err != nil {
+			return nil, apierrors.NewInternalError(fmt.Errorf("failed to update token: %v", err))
+		}
+	}
+
+	return userActivity, nil
+}
+
+// Get implements [rest.Getter]
+func (uas *Store) Get(ctx context.Context,
+	name string,
+	options *metav1.GetOptions) (runtime.Object, error) {
+	return uas.get(ctx, name, options)
+}
+
+// get returns the UserActivity based on the token name.
+// It is used to know, from the frontend, how much time
+// remains before the idle timeout triggers.
+func (uas *Store) get(_ context.Context, uaname string, options *metav1.GetOptions) (runtime.Object, error) {
+	user, token, err := getUserActivityName(uaname)
+	if err != nil {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("wrong useractivity name: %v", err))
+	}
+	// retrieve token information
+	tokenId, err := uas.tokenController.Get(token, *options)
```

Comment:
we have to verify the user doing the request can see this token. Same as https://github.com/rancher/rancher/pull/48778/files#r1962424446

Reply:
addressed in the latest commits